### DISCUSSION TITLE
Access to Private Teams and Project

### DISCUSSION THREAD

**Discussion by magicrocks** [frappe/gameplan#253] [General]
I am using the gameplan and I have created the private teams and private folders. I want to provide access to few users in my team to specific pages and project. I can see the guest  request but can I directly add the internal team members to private teams and project?

**Comment by Senthil455** [comment]
Yes, you can and for internal people you should. Guest flow is not for teammates.
Quick mental model (GP Team = Community, GP Project = Space):
- Gameplan Member/Admin = internal user. Access via GP Member rows on the Team/Space.
- Gameplan Guest = external user. Access via GP Guest Access, only ever per-Space. Guests never join a Community. See gameplan/permissions.py:74-104.
- is_private is what enforces it:
- Private Community (GP Team.is_private=1): only is_community_member() can read. gameplan/permissions.py:89-91
- Private Space (GP Project.is_private=1): only is_space_member() can read. gameplan/permissions.py:102-103
- Public Space in a private Community inherits Community visibility. Public Space in a public Community is open to all Members.

So don't use invite_guest for your own team. That path (GPProject.invite_guest, gameplan/gameplan/doctype/gp_project/gp_project.py:132) creates a GP Invitation with role="Gameplan Guest" -> GP Guest Access, which is intentionally limited to comment/react on that one Space.
For internal users:
1. Make sure they have Gameplan Member role first. If they don't show up in the picker at all, invite them once via Admin invite (gameplan/api.py:125 invite_by_email). Once they have the role, no more email round-trip needed.
2. Private Community: open Community -> Settings/Manage -> Add members. This calls GPTeam.add_members (gameplan/gameplan/doctype/gp_team/gp_team.py:84). You need to be Community admin (require_can_manage_community).
3. Private Space/Page: open Space -> Space access dialog -> Users section (not Guests section) -> Add. This calls GPProject.add_member (gameplan/gameplan/doctype/gp_project/gp_project.py:164). You need to be Space manager (require_can_manage_space_members).

The picker there (frontend/src/components/SpaceAccessDialog.vue:193) only lists Community members, so add to Community first, then to Space.

Two gotchas I hit:

- Removing someone from a private Community also strips their private Space memberships (remove_private_space_memberships, gp_team.py:176), but leaving a public Community via leave_team does not.
- Public Spaces don't have a member list , if you want per-page/per-discussion restriction, the Space itself has to be is_private=1. Content permission just delegates to can_view_space (permissions.py:124).

What you're seeing as "guest request" is just the pending GP Invitation list for externals. Internal Members bypass it entirely with direct add_member(s).
